iothub/device/src/HsmAuthentication/ModuleAuthenticationWithHsm.cs

```diff
@@ -21,25 +21,27 @@ internal class ModuleAuthenticationWithHsm : ModuleAuthenticationWithTokenRefres
         /// <param name="signatureProvider">Provider for the token signature.</param>
         /// <param name="deviceId">Device Identifier.</param>
         /// <param name="moduleId">Module Identifier.</param>
-        public ModuleAuthenticationWithHsm(ISignatureProvider signatureProvider, string deviceId, string moduleId, string generationId) : base(deviceId, moduleId)
+        /// <param name="generationId"></param>
```

Comment:
What should this comment be?

@abhipsaMisra ?

Reply:
"Generation Id for the Edge module"?
Not that I know what it represents, we use it here: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/Edge/EdgeModuleClientFactory.cs#L27

Reply:
https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/Edge/EdgeModuleClientFactory.cs#L96

Reply:
We don't know what generation Id is?

Reply:
I found it looking at the IoT Hub data plane swagger.